src/source.cc

```diff
@@ -2120,7 +2121,25 @@ bool Source::View::on_button_press_event(GdkEventButton *event) {
     get_buffer()->select_range(start, end);
     return true;
   }
-  
+  if((event->type == GDK_BUTTON_PRESS) && (event->button == 3)){
+    auto menu=Menu::get().edit_menu;
```

Comment:
I'm also thinking that we could create a separate menu for the popup on right click, with some selected items from the regular menu (instead of using the edit menu). You can use the same action names as in the regular menu.